lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -397,18 +397,47 @@ void LayoutPropagation::propagateLayout() {
   }
 }
 
+bool isCompatibleLayout(Value v, Attribute layout) {
+  auto tensorTy = dyn_cast<RankedTensorType>(v.getType());
+  if (!tensorTy)
+    return false;
+  auto rank = tensorTy.getRank();
+  auto shape = tensorTy.getShape();
+  if (auto mfma = dyn_cast<AMDMfmaEncodingAttr>(layout)) {
+    if ((shape[rank - 2] != 1 && shape[rank - 2] < mfma.getMDim()) ||
+        (shape[rank - 1] != 1 && shape[rank - 1] < mfma.getNDim())) {
+      return false;
+    }
+  }
+  return true;
+}
```

Comment:
is there any reason why we can't make this layout work?

Reply:
I don't think there are unfixable reason to not make it work "functionally".
But I also do not think we want to use layouts with tiles larger that tensor, because this introduces data and computation duplication.

I had an idea to introduce an heuristic choosing layout based on this data duplication rule, but at this point I am not sure what are the performance implications of this change. Goal of this PR is to fix particular pattern, which almost certainly will drop performance.

Reply:
> But I also do not think we want to use layouts with tiles larger that tensor, because this introduces data and computation duplication.

I think duplication is fine for register-based layouts.

Reply:
> But I also do not think we want to use layouts with tiles larger that tensor, because this introduces data and computation duplication.

it should be better than having extra convert_layout?

Reply:
> it should be better than having extra convert_layout?

In general yes, this is why I am going to try different approach first.

In my particular case, which I am trying to fix, there are no difference in number of `convert_layout` operations.
This is just a matter of where to insert this convert:

case 1 | case 2
--- | ---
%1 = tt.dot tensor 32x32 #mfma32 | %1 = tt.dot tensor 32x32 #mfma32
%2 = reduce %1 tensor 32x1 #mfma32 | %2 = ttg.layout_convert %1 tensor 32x32 #mfma16
%3 = broadcast %2 tensor 32x16 mfma32 | %3 = reduce %2 tensor 32x1 #mfma16
%4 = ttg.layout_convert %3 tensor 32x16 #mfma16 | %4 = broadcast %3 tensor 32x16 mfma16